modules/components/org.wso2.carbon.appfactory.core/src/main/java/org/wso2/carbon/appfactory/core/RemoteRegistryService.java

```diff
@@ -160,4 +162,29 @@ public abstract void copyNonExistingResources(String sourceServerUrl, String sou
 	                                              String destServerUrl, String appId, String username)
 			throws AppFactoryException;
 
+	/**
+	 * Create runtime properties on Kubernetes for a application
+	 *
+	 * @param applicationKey key of the application
+	 * @param version        version of the application
+	 * @param stage          stage of the application
+	 * @param key            key of the property
+	 * @param value          value of the property
+	 * @param isSecured      property is secured or not
+	 * @throws AppFactoryException
+	 */
+	public abstract void addRuntimeProperty(String applicationKey, String version, String stage, String type, String key,
```

Comment:
What is the method argument "type" in here? I believe it is application type. If it is application type, rename it to something like applicationType. Otherwise it hard to identify what the heck is this type. :thumbsdown: